sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java

```diff
@@ -111,8 +111,22 @@ public Plan consume(AnalyzedRelation relation, ConsumerContext context) {
                         joinPushDowns += implicitJoinedRelations.contains(pair) ? 1 : 0;
                     }
                 } else {
-                    // relations are directly joined
-                    joinPushDowns += 1;
+                    // Check if the join condition actually includes both relations
+                    boolean aInJoinCondition = false;
+                    boolean bInJoinCondition = false;
```

Comment:
maybe extract this into a `includesBothRelations(joinPair, a, b) -> bool` method?